src/script/lua_api/l_object.cpp

```diff
@@ -1886,25 +1886,76 @@ int ObjectRef::l_get_sky(lua_State *L)
 	if (player == nullptr)
 		return 0;
 
-	SkyboxParams skybox_params = player->getSkyParams();
+	const SkyboxParams &skybox_params = player->getSkyParams();
+
+	// handle the deprecated version
+	if (!readParam<bool>(L, 2, false)) {
+		log_deprecated(L, "Deprecated call to get_sky, please check lua_api.txt");
+
+		push_ARGB8(L, skybox_params.bgcolor);
+		lua_pushlstring(L, skybox_params.type.c_str(), skybox_params.type.size());
 
+		lua_newtable(L);
+		s16 i = 1;
+		for (const std::string &texture : skybox_params.textures) {
+			lua_pushlstring(L, texture.c_str(), texture.size());
+			lua_rawseti(L, -2, i++);
+		}
+		lua_pushboolean(L, skybox_params.clouds);
+		return 4;
+	}
+
+	lua_newtable(L);
 	push_ARGB8(L, skybox_params.bgcolor);
+	lua_setfield(L, -2, "base_color");
 	lua_pushlstring(L, skybox_params.type.c_str(), skybox_params.type.size());
+	lua_setfield(L, -2, "type");
 
 	lua_newtable(L);
 	s16 i = 1;
 	for (const std::string &texture : skybox_params.textures) {
 		lua_pushlstring(L, texture.c_str(), texture.size());
 		lua_rawseti(L, -2, i++);
 	}
+	lua_setfield(L, -2, "textures");
 	lua_pushboolean(L, skybox_params.clouds);
-	return 4;
+	lua_setfield(L, -2, "clouds");
+
+	lua_newtable(L);
+	if (skybox_params.type == "regular") {
+		push_ARGB8(L, skybox_params.sky_color.day_sky);
+		lua_setfield(L, -2, "day_sky");
+		push_ARGB8(L, skybox_params.sky_color.day_horizon);
+		lua_setfield(L, -2, "day_horizon");
+		push_ARGB8(L, skybox_params.sky_color.dawn_sky);
+		lua_setfield(L, -2, "dawn_sky");
+		push_ARGB8(L, skybox_params.sky_color.dawn_horizon);
+		lua_setfield(L, -2, "dawn_horizon");
+		push_ARGB8(L, skybox_params.sky_color.night_sky);
+		lua_setfield(L, -2, "night_sky");
+		push_ARGB8(L, skybox_params.sky_color.night_horizon);
+		lua_setfield(L, -2, "night_horizon");
+		push_ARGB8(L, skybox_params.sky_color.indoors);
+		lua_setfield(L, -2, "indoors");
+	}
```

Comment:
@Zughy Opinion on this? Seems like the only thing left blocking this change from being merged.

Reply:
I haven't had the chance to look into it yet, but it seems reasonable. I'll push an edit as soon as I can

Reply:
Uhm, wouldn't be weird to have a single private function into a class made only of utilities that will be called from Lua?

Reply:
Sorry, I don't understand your question. I am talking about the duplicated code from L1924 to L1946.
```C++
static void push_sky_color_table(lua_State *L, SkyboxParams &skybox_params)
{
	lua_newtable(L);
	// ... push values
```
It could also be a private class method; there's no functional difference.